### PR TITLE
feat: change to Spontaneous Scope should not be allowed #334

### PR DESCRIPTION
**Pull request by duttarnab**

### CODE REVIEW

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 1]
```diff
@@ -102,6 +132,12 @@ function ScopeForm({ scope, scripts, attributes, handleSubmit }) {
     document.getElementsByClassName('UserActionSubmitButton')[0].click()
   }
 
+  function goToClientViewPage(client_id) {
```
Comment (harryandriyan) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:135]: prefer to use arrow function

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 2]
```diff
@@ -46,6 +65,17 @@ function ScopeForm({ scope, scripts, attributes, handleSubmit }) {
     enableSpontaneous(scope.scopeType),
   )
 
+  useEffect(() => {
+    if (showSpontaneousPanel)
```
Comment (harryandriyan) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:69]: prefer to add a bracket in the if condition `{ ... }`

File: admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js [hunk 3]
```diff
@@ -10,21 +12,38 @@ import {
...
 
-function ScopeForm({ scope, scripts, attributes, handleSubmit }) {
+function ScopeForm({ client, scope, scripts, attributes, handleSubmit, dispatch }) {
   const { t } = useTranslation()
   let dynamicScopeScripts = []
+  const theme = useContext(ThemeContext)
+  const selectedTheme = theme.state.theme
+  const history = useHistory()
   const spontaneousClientScopes = scope.attributes.spontaneousClientScopes || []
+  const options = {}
```
Comment (harryandriyan) [inline admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js:46]: It's better to use `useState` 
```JS
const [options, setOptions] = useState();

// And in the makeOptions function we cal setOptions
...

const makeOption = (limit, client_id) => {
  setOptions({ [LIMIT]: limit, [PATTERN]: client_id });
}

```

**Review by mjatin-dev — APPROVED** [review]
Reviewed by Harry

**Pull request merged**